### ISSUE TITLE
Undocumented JSON formatting change in v2.4

### ISSUE THREAD

**Issue by gustafsonk** [IdentityServer/IdentityServer3#2822]
When requesting an access token via the /connect/token endpoint, the raw JSON response now has whitespace and newlines, which I think is a result of #2382 . I think reverting to the previous behavior should be easy by changing line 86 of https://github.com/IdentityServer/IdentityServer3/blob/8912c77d8dfb48e6f7565273e7d2d099114dc86b/source/Core/Results/TokenResult.cs from jObject.ToString() to jObject.ToString(Formatting.None)

**Comment by leastprivilege** [comment]
Hi, 

this is possible (haven't checked yet) - but why does it matter. Did the semantics of the JSON payload change - or just the formatting?

**Comment by gustafsonk** [comment]
It's just the formatting that changed. For many people this doesn't matter because they are using good JSON parsing libraries that know how to handle whitespace. In fact, the only reason we noticed it was because we have one team who decided to use a crappy JSON parser in a product and now we can't upgrade our production environment beyond v2.3 without pushing out a firmware update to these devices.

I guess another reason to change it back would be to reduce the response size ever so slightly to save on bandwidth costs.

I would really appreciate it if this was reverted back since it would mean I don't need to fork IdSrv to use the cool new features in v2.5. Thanks.

**Comment by leastprivilege** [comment]
OK - I will look into it.

**Comment by leastprivilege** [comment]
OK - schedule for 2.5.1

**Comment by leastprivilege** [comment]
done on dev

**Comment by CrescentFresh** [comment]
Oh this should not have been reverted.

We had the same issue as gustafsonk with one of our customers: they were doing substring indexing to gain the access token from the json response; newlines broke their production code. 

However instead of asking for you to revert the change, we told them to fix their damn code.

However, we are not sure if they introduced proper json parsing (which believe me we recommended they do) or simply changed their index numbers.

So now with this revert we may be back to the same issue again ironically.

Just my two cents.
